torch/csrc/utils/python_symnode.h

```diff
@@ -32,27 +33,37 @@ namespace impl {
 // this is just an adapter so C++ calls can get to the object.
 class PythonSymNodeImpl : public c10::SymNodeImpl {
  public:
-  PythonSymNodeImpl(py::object pyobj) : c10::SymNodeImpl() {
+  PythonSymNodeImpl(
+      py::object pyobj,
+      c10::optional<bool> is_singleton = c10::nullopt)
```

Comment:
Or document that this is a performance optimization, and more clearly explain what the correctness condition for calling it is

Reply:
just axe it lol